pkg/helm/index.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright © 2021 VMware
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package helm
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+
+	"github.com/ghodss/yaml"
+	"github.com/jinzhu/copier"
+	"github.com/kubeapps/kubeapps/pkg/chart/models"
+	helmrepo "k8s.io/helm/pkg/repo"
+	log "k8s.io/klog/v2"
+)
+
+func parseRepoIndex(contents []byte) (*helmrepo.IndexFile, error) {
+	var index helmrepo.IndexFile
+	err := yaml.Unmarshal(contents, &index)
+	if err != nil {
+		return nil, err
+	}
+	index.SortEntries()
+	return &index, nil
+}
+
+// Takes an entry from the index and constructs a database representation of the
+// object.
+func newChart(entry helmrepo.ChartVersions, r *models.Repo, shallow bool) models.Chart {
+	var c models.Chart
+	copier.Copy(&c, entry[0])
+	if shallow {
+		copier.Copy(&c.ChartVersions, []helmrepo.ChartVersion{*entry[0]})
+	} else {
+		copier.Copy(&c.ChartVersions, entry)
+	}
+	c.Repo = r
+	c.Name = url.PathEscape(c.Name) // escaped chart name eg. foo/bar becomes foo%2Fbar
+	c.ID = fmt.Sprintf("%s/%s", r.Name, c.Name)
+	c.Category = entry[0].Annotations["category"]
+	return c
+}
+
+func ChartsFromIndex(contents []byte, r *models.Repo, shallow bool) ([]models.Chart, error) {
```

Comment:
In exported (at least) functions I'd like to have some documentation around it. Perhaps this PR is just about moving the existing code, but since it's now public, I think it might be worthwhile to add a couple of lines, no?

https://blog.golang.org/godoc

Reply:
sure will do